extension.js

```diff
@@ -36,7 +36,15 @@ class ExtensionDBusInterface {
             return;
 
         Main.wm.skipNextEffect(current_window.get_compositor_private());
+
+        const was_maximized = settings.get_boolean('window-maximize');
         current_window.unmaximize(Meta.MaximizeFlags.VERTICAL);
+
+        if (was_maximized) {
```

Comment:
I still think this `if` is not necessary. Window was maximized, mouse resizing starts - it means that the window should occupy the entire workarea, regardless of `window-maximize`